man/systemd-nspawn.xml

```diff
@@ -1352,6 +1352,58 @@ After=sys-subsystem-net-devices-ens1.device</programlisting>
+
+        <listitem><para>Binds the home directory of the specified user on the host into the container. Takes
+        the name of an existing user on the host as argument. May be used multiple times to bind multiple
+        users into the container. This does three things:</para>
+
+        <orderedlist>
+          <listitem><para>The user's home directory is bind mounted from the host into
+          <filename>/run/hosts/home/</filename>.</para></listitem>
+
+          <listitem><para>An additional UID/GID mapping is added that maps the host user's UID/GID to a
+          container UID/GID, allocated from the 60514…60577 range.</para></listitem>
+
+          <listitem><para>A JSON user and group record is generated in <filename>/run/userdb/</filename> that
+          describes the the mapped user. It contains a minimized representation of the host's user record,
+          adjusted to the UID/GID and home directory path assigned to the user in the container. The
+          <citerefentry><refentrytitle>nss-systemd</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+          glibc NSS module will pick up these records from there and make them available in the containers
+          user/group databases.</para></listitem>
+        </orderedlist>
+
+        <para>The combination of the three operations above ensures that it is possible to log into the
+        host's user account inside the container as if it was local to the container. The user is only mapped
+        transiently, while the container is running and the mapping itself does not result in persistent
+        changes to the container (except maybe for generated log messages at login time, and similar). Note
+        that in particular the UID/GID assignment in the container is not made persistently. Thus: if a user
+        after logging in makes changes to the image (outside of their home directory or transient
+        directories), and leaves files or directories around that are owned by them, they will remain
+        associated with the used UID/GID. As these UID/GIDs might be recycled during later container
+        invocations (with possibly different <option>--bind-user=</option> settings) and be assigned to
+        different users or groups, this is potentially a security issue.</para>
+
+        <para>The user/group record mapping only works if the container contains systemd 249 or newer, with
+        <command>nss-systemd</command> properly configured in <filename>nsswitch.conf</filename>. See
+        <citerefentry><refentrytitle>nss-systemd</refentrytitle><manvolnum>8</manvolnum></citerefentry> for
+        details.</para>
+
+        <para>Note that the user record propagated from the host into the container will contain the UNIX
+        password hash of the user, so that seamless logins in the container are possible. If the container is
+        less trusted than the host it's hence important to use a strong UNIX password hash function.</para>
```

Comment:
Uuuuh, I don't think "strong UNIX password hash function" is something that exists in this sense. A strong hash raises a bar a bit, but with rainbow tables and whatnot it's not enough of a bar. So if the container payload is untrusted, `--bind-user` must not be used, period.

Reply:
OK, I guess yescrypt or argon2 should be OK. But the default on Fedora is sha512… Please confirm, but my understanding is that it has weaker properties. Also, even if the *current* default is updated, people might legitimately have older passwords. Most people would interpret sha-* as "strong".

>  I mean, a USB stick with a LUKS volume on it is not really better protected...

Possibly, but it's not relevant, because the the exposure is different. The usb stick is protected by physical security, and we're not allowing direct access to the device to untrusted components. And we're discussing a container payload that is untrusted.

I don't think we should get into the business of evaluating attack scenarios. Maybe just say:
> "Note that the user record propagated from the host into the container will contain the UNIX password hash of the user,
> to allow password-based authentication. This potentially allows attacks that subvert the hashing function to derive a matching password. If the container payload is not fully trusted, this might be mitigated using a strong-enough password hashing function (yescrypt or argon2), or disallowing exfiltration of data from the container, etc.

(OK, I see that my text is not good. It's hard to write something that is concise.)